### PR TITLE
Unit test backdrops controller

### PR DESCRIPTION
**Pull request by alexanderglenn**
The spec tesst for the backdrops_controller.rb . There are 27 tests in total. The tests cover all the public methods within the controller apart from the 'edit' method which is empty.
To run the tests use: rspec limelight/spec/controllers/backdrops_controller_spec.rb
To run an individual test use the above command with the line number. E.G. rspec limelight/spec/controllers/backdrops_controller_spec.rb:35
unit-test-cap

### CODE REVIEW

File: spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  #     --seed 1234
+  config.order = :random
+
+  # Seed global randomization in this process using the `--seed` CLI option.
+  # Setting this allows you to use `--seed` to deterministically reproduce
+  # test failures related to randomization by passing the same `--seed` value
+  # as the one that triggered the failure.
+  Kernel.srand config.seed
+=end
+end
```
Comment (charlie0605) [inline spec_helper.rb]: why is there 2 spec helper file, one at the root of the project and one in the spec file

File: support/factory_girl.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# spec/support/factory_girl.rb
+
+RSpec.configure do |config|
+	config.include FactoryGirl::Syntax::Methods
+end
```
Comment (charlie0605) [inline support/factory_girl.rb]: Why is this added? I don't see you have used FactoryGirl anywhere in your test.

File: spec/controllers/backdrops_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+	end
+
+
+	describe "POST #create" do
+		context "Valid backdrop" do
+
+			before :each do
+				user = User.create({ nickname: 'Admin', email: 'admin@local.instance', password: 'admin', password_confirmation: 'admin', is_active: true, email_confirmed: Time.zone.now })
+				cookies[:auth_token] = user.auth_token				
+			end
```
Comment (charlie0605) [inline spec/controllers/backdrops_controller_spec.rb]: There's too many before :each method, this can be reduced

File: spec/controllers/backdrops_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+	 		end
+	 	end
+
+	 	context "Invalid backdrop" do
+
+	 		before :each do
+	 			user = User.create({ nickname: 'Admin', email: 'admin@local.instance', password: 'admin', password_confirmation: 'admin', is_active: true, email_confirmed: Time.zone.now })
+			 	cookies[:auth_token] = user.auth_token
+			 	post :create, :params => { :backdrop => { name: "" } }
+	 		end
```
Comment (charlie0605) [inline spec/controllers/backdrops_controller_spec.rb]: another one here

File: spec/controllers/backdrops_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+	 	 end
+	 end
+
+	describe "#show" do
+
+		subject { post :create, :params => { :backdrop => { name: "Pass Type", source_name: "bg1", source_content_type: "image/png" } } }
+
+	 	it "should redirect to edit backdrop path" do
+	 		user = User.create({ nickname: 'Admin', email: 'admin@local.instance', password: 'admin', password_confirmation: 'admin', is_active: true, email_confirmed: Time.zone.now })
+			cookies[:auth_token] = user.auth_token
```
Comment (charlie0605) [inline spec/controllers/backdrops_controller_spec.rb]: another one here

File: spec/controllers/backdrops_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+			expect(subject).to redirect_to(edit_backdrop_path(Backdrop.last))
+	 	end
+	end
+
+	describe "#PATCH update" do
+		context "stage is present" do
+
+			before :each do
+				user = User.create({ nickname: 'Admin', email: 'admin@local.instance', password: 'admin', password_confirmation: 'admin', is_active: true, email_confirmed: Time.zone.now })
+				cookies[:auth_token] = user.auth_token
```
Comment (charlie0605) [inline spec/controllers/backdrops_controller_spec.rb]: another one here

File: spec/controllers/backdrops_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+				expect(response).to redirect_to(@stage)
+			end
+		end
+
+		context "stage not present" do
+			context "@backdrop.update(backdrop_params)" do
+				before :each do
+					@backdrop = Backdrop.create(name: "bg1")
+					user = User.create({ nickname: 'Admin', email: 'admin@local.instance', password: 'admin', password_confirmation: 'admin', is_active: true, email_confirmed: Time.zone.now })
+					cookies[:auth_token] = user.auth_token
```
Comment (charlie0605) [inline spec/controllers/backdrops_controller_spec.rb]: another one here

File: spec/controllers/backdrops_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+
+				it "should change the name to the new name" do
+					expect(@backdrop.name).to eq("New name")
+				end
+			end
+
+			context "@backdrop.update(backdrop_params) eq false" do
+				before :each do
+					@backdrop = Backdrop.create(name: "bg1")
+					user = User.create({ nickname: 'Admin', email: 'admin@local.instance', password: 'admin', password_confirmation: 'admin', is_active: true, email_confirmed: Time.zone.now })
```
Comment (charlie0605) [inline spec/controllers/backdrops_controller_spec.rb]: another one here

File: spec/controllers/backdrops_controller_spec.rb [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+				end
+			end
+		end
+	end
+
+	describe "DELETE destroy" do
+		context "stage is present" do
+			before :each do
+				user = User.create({ nickname: 'Admin', email: 'admin@local.instance', password: 'admin', password_confirmation: 'admin', is_active: true, email_confirmed: Time.zone.now })
+				cookies[:auth_token] = user.auth_token
```
Comment (charlie0605) [inline spec/controllers/backdrops_controller_spec.rb]: another one here

File: spec/controllers/backdrops_controller_spec.rb [hunk 10]
```diff
@@ -0,0 +1,241 @@
...
+			end
+
+			it "should redirect to the stage" do
+				expect(response).to redirect_to(@stage)
+			end
+		end
+
+		context "stage not present" do		
+
+			context "@backdrop.destroy" do
```
Comment (charlie0605) [inline spec/controllers/backdrops_controller_spec.rb]: Please check the usage of describe and context
have a look at this link: http://www.betterspecs.org/

File: spec/controllers/backdrops_controller_spec.rb [hunk 11]
```diff
@@ -0,0 +1,241 @@
...
+				before :each do
+					user = User.create({ nickname: 'Admin', email: 'admin@local.instance', password: 'admin', password_confirmation: 'admin', is_active: true, email_confirmed: Time.zone.now })
+					cookies[:auth_token] = user.auth_token
```
Comment (charlie0605) [inline spec/controllers/backdrops_controller_spec.rb]: more user created here

File: spec/controllers/backdrops_controller_spec.rb [hunk 12]
```diff
@@ -0,0 +1,241 @@
...
+				end
+				it "should redirect to backdrops path" do
+					expect(response).to redirect_to(backdrops_path)
+				end
+			end
+
+			context "@backdrop.destroy eq false" do
+				before :each do
+					user = User.create({ nickname: 'Admin', email: 'admin@local.instance', password: 'admin', password_confirmation: 'admin', is_active: true, email_confirmed: Time.zone.now })
+					cookies[:auth_token] = user.auth_token
```
Comment (charlie0605) [inline spec/controllers/backdrops_controller_spec.rb]: here also

**Comment by rohrlach** [comment]
@alexanderglenn I suggest that you resolve the conflicts listed and update your branch.
